### DISCUSSION TITLE
module declares its path as: cloud.google.com/go but was required as: google.golang.org/cloud issue

### DISCUSSION THREAD

**Discussion by developer-guy** [distribution/distribution#4404] [General]
I was trying to compile the project, so, I cloned the project and did check-out to tags/v2.8.3 (which is the latest at the time of writing this) then I noticed there is no go.mod or go.sum files so I did `go mod init` and `go mod tidy` afterwards, this is the issue I had:

```bash
...
2024/07/18 08:05:55 WARN go: github.com/distribution/distribution/v3/registry/storage/driver/gcs imports
2024/07/18 08:05:55 WARN     google.golang.org/cloud: google.golang.org/cloud@v0.115.0: parsing go.mod:
2024/07/18 08:05:55 WARN     module declares its path as: cloud.google.com/go
2024/07/18 08:05:55 WARN             but was required as: google.golang.org/cloud
```

btw I also tried to download the exist go.mod file from main and re-run go mod tidy for the given tag, it didn't work either, so any thoughts on this?

Thank you

**Comment by milosgajdos** [comment]
Please do not use the `v2` release/code. It will be deprecated once the stable v3 release is made. It's effectively in a maintenance mode at this point even though the README does not state so clearly enough.

Yes, the v2 tag code does not use `go.mod`; it's an ancient codebase that wasn't worth migrating.

**Reply by developer-guy** [reply]
gotcha, thank you so much for such a quick response!

**Reply by sgalsaleh** [reply]
@milosgajdos I'm testing out [v3.0.0-beta.1](https://github.com/distribution/distribution/releases/tag/v3.0.0-beta.1), is it possible to disable OpenTelemetry? logs are filled with:
```bash
level=error msg="traces export: Post \"https://localhost:4318/v1/traces\": dial tcp [::1]:4318: connect: connection refused"
```

**Reply by milosgajdos** [reply]
@sgalsaleh 👉 https://github.com/distribution/distribution/issues/4270
